product-catalog/app/com/commercetools/sunrise/productcatalog/productdetail/ProductDetailPageContentFactory.java

```diff
@@ -22,12 +22,15 @@
     protected ProductBeanFactory productBeanFactory;
 
     public ProductDetailPageContent create(final ProductProjection product, final ProductVariant variant) {
-        final ProductDetailPageContent content = new ProductDetailPageContent();
-        fillAdditionalTitle(product, content);
-        fillProduct(product, variant, content);
-        fillBreadCrumb(product, variant, content);
-        fillAddToCartFormUrl(content);
-        return content;
+        return fillBean(new ProductDetailPageContent(), product, variant);
+    }
+
+    protected <T extends ProductDetailPageContent> T fillBean(final T bean, final ProductProjection product, final ProductVariant variant) {
+        fillAdditionalTitle(product, bean);
+        fillProduct(product, variant, bean);
+        fillBreadCrumb(product, variant, bean);
+        fillAddToCartFormUrl(bean);
```

Comment:
For me it would be either:
```Java
protected <T extends LineItemBean> T fillBean(final T bean, LineItem lineItem)
```
or
```Java
protected <T extends LineItemBean> void fillBean(final T bean, LineItem lineItem)
```

I don't have a strong opinion. I prefer the one without side-effects in general, but in this particular case we are dealing with Java Beans, so side-effects is everything we have, so it makes sense here.
I leave the decision to you :bowtie: